### ISSUE TITLE
Inner return calls fail the require-returns rule

### ISSUE THREAD

**Issue by swcisel** [gajus/eslint-plugin-jsdoc#151]
The following function has no returned value, but it is marked as failing the `require-returns` rule because of the internal `return` call from the array's `filter()` method:

```
removeItem(itemForRemoval) {
    const newItems = this.Items.filter(
      ({ label, index }) => {
        return label !== itemForRemoval.label;
      }
    );
    this.saveItems(newItems);
 }
```

**Comment by kewisch** [comment]
This also happens with an arrow function:

```javascript
/**
 * A test function that has an arrow function.
 */
function foo() {
  data.map(bar => bar.baz);
}
```

I'd argue this is a pretty common case, would be great to have this fixed.

**Comment by brettz9** [comment]
The example in the original post is not reporting an issue, though the arrow function is. Fixed by: #217
